ls1intum/Artemis
# `General`: Fix statistics view showing incorrect scores

### Checklist
#### General


#### Client

### Motivation and Context


Fixes #6254 
Fixes #5742 
Fixes #6103 
Fixes #6263

### Description

If the cached participations are empty, they overwrite the actual student participations received from the server. 
A check was added, that only overwrites the student participation in case the cached participation is not empty.

### Steps for Testing

Prerequisites:
- 1 Student
- 1 Programming Exercise with more than 0 points achieved

1. Log in to Artemis
2. Navigate to the Course Overview & click on a course
3. Click on Statistics
4. Make sure the overall points are displayed correctly (for reference compare it to the screen recording below)
5. Reload the page / Click the refresh button and make sure the scores stay the same

### Review Progress






#### Code Review
#### Manual Tests

### Test Coverage





### Screenshots



Before:

https://user-images.githubusercontent.com/9954674/219817460-4f95caa1-a3ab-4e43-bf5d-e4cb8205471d.mov

---
After:

https://user-images.githubusercontent.com/9954674/219817225-27b846d8-b4c1-4158-99b7-dfc4680488c5.mov
